crates/core/src/lib.rs

```diff
@@ -399,6 +399,14 @@ impl<T: OutboundWasiHttpHandler + Send + Sync> Engine<T> {
         Ok(InstancePre { inner })
     }
 
+    /// Returns the type of this component to reflect on it at runtime.
+    #[instrument(skip_all)]
+    pub fn component_type(&self, component: &Component) -> Result<ComponentType> {
+        // NB: With Wasmtime 20 and bytecodealliance/wasmtime#8078 this method
+        // should probably be removed to just use that instead.
+        self.linker.substituted_component_type(component)
+    }
+
     /// Creates a new [`ModuleInstancePre`] for the given [`Module`].
     #[instrument(skip_all)]
```

Comment:
...though I'm not sure why this one is instrumented either... :thinking:

Reply:
Ah it used to acquire a lock I was probably concerned about. Could probably be removed now. :shrug: